R/known-sources.R

```diff
@@ -104,6 +104,40 @@ taxi_schema <- function() {
 }
 
 known_datasets <- list(
+  taxi_2013 = list(
+    url = "https://archive.org/download/nycTaxiTripData2013/trip_fare.7z",
+    download = function(path) {
+      archive_path <- file.path(dirname(path), "trip_fare.7z")
+      if (!file.exists(archive_path)) {
+        download.file(
+          "https://archive.org/download/nycTaxiTripData2013/trip_fare.7z",
+          archive_path,
+          mode = 'wb',
+          method = 'wget'
+        )
+      }
+
+      archive::archive_extract(archive_path, dir = path)
+
+      files <- list.files(path, pattern = '\\.csv$', full.names = TRUE)
+      lapply(files, function(file) {
+        message("Fixing headers in ", file)
+        lines <- readr::read_lines(file)
+        lines[[1]] <- gsub(", ", ",", lines[[1]], fixed = TRUE)
+        readr::write_lines(lines, file)
```

Comment:
I'm fine leaving this as is, though maybe a comment here explaining the hack + that we could do something like `sed` which is _much quicker_ but is more complicated in a crossplatform way

Reply:
I'ld need to dig it up in the git history, but I _did_ at some point have some code that switched `sed` based on OS for this purpose. It was pretty hacky though and I was glad to get rid of it when I could, so not sure we want to model that necessarily.

Reply:
Yeah I really don't like this solution—it may very well screw with line endings or encoding or whatever, plus it's slow—but I couldn't find a good alternative. It looks like it's technically possible (in C++ at least) to mutate a file in place which would be the least-invasive way to fix these (even `sed` copies to a buffer), but mostly nobody does that because if you screw it up your file is borked.

Reply:
Absolutely — if you don't have it already, let's add a comment that basically says that and we can merge